mesa/space.py

```diff
@@ -548,7 +550,428 @@ def exists_empty_cells(self) -> bool:
         return len(self.empties) > 0
 
 
-class SingleGrid(_Grid):
+def is_lambda_function(function):
```

Comment:
Maybe call it is_single_argument_function!?

Reply:
I would agree we want to try an consolidate these into one function. As you call it numerous times

Reply:
https://github.com/projectmesa/mesa/pull/1898/commits/6e12379dc1194522be515c7373974a2d79016b7f